azurelinuxagent/ga/exthandlers.py

```diff
@@ -298,11 +298,28 @@ def run(self):
             self.log_report = True
 
             if self._extension_processing_allowed() and self._incarnation_changed(etag):
-                activity_id, correlation_id, gs_creation_time = self.get_goal_state_debug_metadata()
-
-                logger.info(
-                    "ProcessGoalState started [Incarnation: {0}; Activity Id: {1}; Correlation Id: {2}; GS Creation Time: {3}]".format(
-                        etag, activity_id, correlation_id, gs_creation_time))
+                try:
+                    activity_id, correlation_id, gs_creation_time = self.get_goal_state_debug_metadata()
+
+                    debug_info = "; ".join([
+                        "Incarnation: {0}".format(etag),
+                        "Activity Id: {0}".format(activity_id),
+                        "Correlation Id: {0}".format(correlation_id),
+                        "GS Creation Time: {0}".format(gs_creation_time)
+                    ])
+                    logger.info("ProcessGoalState started [{0}]", debug_info)
+                except Exception as e:
+                    # A failure here isn't a fatal error, because the info we're
+                    # trying to retrieve is debug only on linux. We still need to 
+                    # process the goal state.
+                    error_msg = u"ProcessGoalState started, but failed to retrieve gs debug metadata: '{0}'"\
```

Comment:
I think this is a bit too much debug data for something small (i.e. debug data). If this data is not there or we're unable to parse it properly, we can just skip it altogether and just log a basic message. At max, just log it once as a warn or something per goalstate. No need for Kusto logging here either.